### ISSUE TITLE
Verbose Voiceover when switching focus

### ISSUE THREAD

**Issue by sana-ajani** [microsoft/vscode#88727]
Voiceover in VS Code is very chatty and verbose, compared to MacDown. In VS Code, it reads a lot of the internal constructs and extraneous info. Create a new blank file in VS Code. Switch focus away from VS Code, and re-enter VS Code. "Untitled" file name is repeated multiple times, and the usage of "Group" is misleading, "Group" in voiceover refers to a group of buttons/options to further go into, but I think we are using it to mean editor groups. @ivanfetch 

![image](https://user-images.githubusercontent.com/9721254/72481623-f6f0c480-37af-11ea-8ea4-685564636eb7.png)

Compared to MacDown (markdown editor in macos)

![image](https://user-images.githubusercontent.com/9721254/72481639-01ab5980-37b0-11ea-9620-22ac5aa4ebe3.png)

Version: 1.42.0-insider
Electron: 7.1.7
OS: MacOS Mojave

**Comment by isidorn** [comment]
I can reproduce this, and this is a valid issue. I think that Voice Over reads the title area of VS Code and than the aria label of the Editor area. However I get this chatty text every time I focus on an Editor Group.
So we should be more concise, and we sohuld say `Editor Group` instead of `Group`.
And we could remove "Untitled file text editor", that is already pretty obvious from the title.

@bpasero what do you think?

**Comment by bpasero** [comment]
But removing that, wouldn't it mean we no longer read the editor type when switching focus to another editor?

**Comment by bpasero** [comment]
I am not sure how useful it is, you would have to ask someone that relies on this information. Feel free to change it.

**Comment by isidorn** [comment]
This is comming directly from a user that uses VoiceOver. Yeah I will change it and try to make it more concise.

**Comment by bpasero** [comment]
I think this is all coming as option into the editor via [`getAriaLabel`](https://github.com/microsoft/vscode/blob/11932a9ef46ec13bdcf293d1fa3a0c82409bfa3a/src/vs/workbench/browser/parts/editor/textEditor.ts#L304)

**Comment by isidorn** [comment]
@ivanfetch based on your feedback I have changed this in the following way:

Before it was reading:
"Untitled-1 Untitled file text editor, Group 1, edit text"

Now it is reading:
"Untitled-1 editor, Editor Group 1, edit text"

Now we are trying to be more concise, and we use the term Editor Group instead of Group which is VoiceOver specific.

@ivanfetch can you please try it out with vscode insiders from Tuesday and let us know how it works for you now. If there are still things to polish let us know and we can reopen this.

@bpasero even though you might change that the untitled editors no longer have the Untitled as the name I think that is fine since that information is not that cruical to hear every time we focus. Also note that I introduced an `ariaLabel` field to the `editorGroup`

**Comment by bpasero** [comment]
@isidorn I see in https://github.com/microsoft/vscode/commit/9bf0af3ea0715d73cfa6440a14f4487028412d74 that the new group.ariaLabel property is only used for the text editor, isn't that a bit inconsistent? What about all the other editors? I am also easy on dropping the aria label for the group altogether if that is helpful.

**Comment by bpasero** [comment]
Oh nevermind, I see that we use it consistently from the `computeAriaLabel`.

**Comment by bpasero** [comment]
I have fixed https://github.com/microsoft/vscode/issues/88981 and pushed another change to properly indicate readonly status for file editors (something we do for all other editors but not for file editors so far).

**Comment by isidorn** [comment]
@bpasero thanks a lot!
@ivanfetch please note that we still read the title and the editor aria when you focus into VS Code resulting in some duplication, however we should behave better now overall. I am open for suggestions on how to fix this.

**Comment by ivanfetch** [comment]
The title of the edit area is less verbose - compared to what you described in your comment above, there is an extra `group` being spoken, and the first line of text in the edit is read.

Using the insiders build which I've just downloaded, - entering the edit area with Voiceover now reads `x editor, Editor Group 1 group`. The `x` above represents the first line of text typed into the edit area - this is spoken if I use VoiceOver to navigate into, or out of, the edit area. The first line of text is spoken no matter what line my cursor is on.

**Comment by isidorn** [comment]
@ivanfetch also please set the following setting
`"window.title": "\u200B"` That way the window title will be empty and there will be no duplication when the VS Code window gets focused.
We were thinking about automatically changing this when a screen reader is detected however we decided that behavior is not optimal since third party apps might depend on the Title. More details here https://github.com/microsoft/vscode/pull/89216

I can reproduce the group and the first line of text. To have a clear slate to start a discussion I created two issues,:
https://github.com/microsoft/vscode/issues/89231
https://github.com/microsoft/vscode/issues/89230
